text/profiles/0239-profiles-data-model.md

```diff
@@ -0,0 +1,1610 @@
+
+<details>
+<summary>Field Descriptions</summary>
+
+##### Field `profile_id`
+
+A unique identifier for a profile. The ID is a 16-byte array. An ID with
+all zeroes is considered invalid.
+This field is required.
+
+##### Field `start_time_unix_nano`
+
+start_time_unix_nano is the start time of the profile.
+Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+This field is semantically required and it is expected that end_time >= start_time.
+
+##### Field `end_time_unix_nano`
+
+end_time_unix_nano is the end time of the profile.
+Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
+This field is semantically required and it is expected that end_time >= start_time.
+
+##### Field `attributes`
+
+attributes is a collection of key/value pairs. Note, global attributes
+like server name can be set using the resource API.
+
+##### Field `dropped_attributes_count`
+
+dropped_attributes_count is the number of attributes that were discarded. Attributes
+can be discarded because their keys are too long or because there are too many
+attributes. If this value is 0, then no attributes were dropped.
+
+##### Field `original_payload_format`
+
+Specifies format of the original payload. Common values are defined in semantic conventions. [required if original_payload is present]
+
+##### Field `original_payload`
+
+Original payload can be stored in this field. This can be useful for users who want to get the original payload.
+Formats such as JFR are highly extensible and can contain more information than what is defined in this spec.
+Inclusion of original payload should be configurable by the user. Default behavior should be to not include the original payload.
+If the original payload is in pprof format, it SHOULD not be included in this field.
+The field is optional, however if it is present `profile` MUST be present and contain the same profiling information.
+
+##### Field `profile`
+
+This is a reference to a pprof profile. Required, even when original_payload is present.
+</details>
+
+#### Message `Profile`
+
+Profile is a common stacktrace profile format.
+Measurements represented with this format should follow the
+following conventions:
+
+- Consumers should treat unset optional fields as if they had been
+  set with their default value.
+- When possible, measurements should be stored in "unsampled" form
+  that is most useful to humans.  There should be enough
+  information present to determine the original sampled values.
+- On-disk, the serialized proto must be gzip-compressed.
+- The profile is represented as a set of samples, where each sample
+  references a sequence of locations, and where each location belongs
+  to a mapping.
+- There is a N->1 relationship from sample.location_id entries to
+  locations. For every sample.location_id entry there must be a
+  unique Location with that id.
+- There is an optional N->1 relationship from locations to
```

Comment:
This is also what we show in the ASCII diagram

```suggestion
- There is an optional 1->1 relationship from locations to
```

Reply:
I think the diagram is incorrect, I'll fix.